_search-plugins/knn/api.md

```diff
@@ -156,51 +152,97 @@ GET /_plugins/_knn/warmup/index1,index2,index3?pretty
-training step before indexing and querying can begin. The output of training is a model that can then be used to 
-initialize native library index files during indexing. The model is serialized in the k-NN model system index.  
+During approximate k-NN search or warmup operation, the native library indexes (`nmslib` and `faiss` engines) are loaded into native memory. Currently, you can evict an index from cache or native memory by either deleting the index or setting the k-NN cluster settings `knn.cache.item.expiry.enabled` and `knn.cache.item.expiry.minutes`, which removes the index out of the cache if they are idle for a given period of time. However, you cannot evict an index from cache without deleting the index. To solve this problem, you can use the k-NN clear cache API operation that clears a given set of indexes from the cache.
+
+The k-NN clear cache API evicts all native library files for all shards (primaries and replicas) of all indexes specified in the request. Similar to how the [warmup operation](#warmup-operation) behaves, the k-NN clear cache API is idempotent, meaning that if you try to clear the cache for an index that has already been evicted from the cache, it does not have any additional impact or effect.
+
+This API operation works only with indexes created using the `nmslib` and `faiss` engines. The API operation has no impact on the indexes created using `lucene` engine.
+{: .note}
+
+#### Usage
+
+The following request evicts the native library indexes of three indexes from the cache:
+
+```json
+GET /_plugins/_knn/clear_cache/index1,index2,index3?pretty
+{
+  "_shards" : {
+    "total" : 6,
+    "successful" : 6,
+    "failed" : 0
+  }
+}
+```
+
+The `total` parameter indicates number of shards the API attempted to clear from the cache. The response includes the number of shards the plugin succeeded and failed to clear.
+
+k-NN clear cache API can be used with index patterns to clear one or more indexes that match the given pattern from the cache, as shown in the following example:
+
+```json
+GET /_plugins/_knn/clear_cache/index*?pretty
+{
+  "_shards" : {
+    "total" : 6,
+    "successful" : 6,
+    "failed" : 0
+  }
+}
+```
+
+The API call does not return results until the operation finishes or the request times out. If the request times out, the operation continues on the cluster. To monitor the request, use the `_tasks` API, as shown in the following example:
+
+```json
+GET /_tasks
+```
+
+When the operation finishes, use the [k-NN `_stats` API operation](#stats) to view which indexes have been evicted from the cache.
+
+## Get model
+
+The get model retrieves information about models present in the cluster. Some native library index configurations require a training step before indexing and querying can begin. The output of training is a model that can be used to initialize native library index files during indexing. The model is serialized in the k-NN model system index. See the following GET example:
```

Comment:
The beginning of this sentence is missing words. "The `get model` operation" or "The GET model operation"? Also ensure that the heading matches, as appropriate.

Reply:
@natebower After reviewing your comment, I agree this section needs some refinement. See my revisions.

Reply:
@vagimeli LGTM 😄